src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -5116,6 +5116,368 @@ .maxstack  2
+    }
+
+    static int M(string x, int y)
+    {
+        return (x, y) switch
+        {
+            ("""", 0) => 0,
+            ("""", 1) => 1,
+            (_, 2) => 2,
+            (_, 3) => 3,
+            (_, 4) => 4,
+            _ => 5
+        };
+    }
+}
+";
+            var verifier = CompileAndVerify(source, expectedOutput: "0123452345", options: TestOptions.DebugExe);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("Program.M", @"{
+  // Code size       98 (0x62)
+  .maxstack  2
+  .locals init (int V_0,
+                int V_1)
+  IL_0000:  nop
+  IL_0001:  ldc.i4.1
+  IL_0002:  brtrue.s   IL_0005
+  IL_0004:  nop
+  IL_0005:  ldarg.0
+  IL_0006:  ldstr      """"
+  IL_000b:  call       ""bool string.op_Equality(string, string)""
+  IL_0010:  brfalse.s  IL_002e
+  IL_0012:  ldarg.1
+  IL_0013:  switch    (
+        IL_0040,
+        IL_0044,
+        IL_0048,
+        IL_004c,
+        IL_0050)
+  IL_002c:  br.s       IL_0054
+  IL_002e:  ldarg.1
+  IL_002f:  ldc.i4.2
+  IL_0030:  beq.s      IL_0048
+  IL_0032:  br.s       IL_0034
+  IL_0034:  ldarg.1
+  IL_0035:  ldc.i4.3
+  IL_0036:  beq.s      IL_004c
+  IL_0038:  br.s       IL_003a
+  IL_003a:  ldarg.1
+  IL_003b:  ldc.i4.4
+  IL_003c:  beq.s      IL_0050
+  IL_003e:  br.s       IL_0054
+  IL_0040:  ldc.i4.0
+  IL_0041:  stloc.0
+  IL_0042:  br.s       IL_0058
+  IL_0044:  ldc.i4.1
+  IL_0045:  stloc.0
+  IL_0046:  br.s       IL_0058
+  IL_0048:  ldc.i4.2
+  IL_0049:  stloc.0
+  IL_004a:  br.s       IL_0058
+  IL_004c:  ldc.i4.3
+  IL_004d:  stloc.0
+  IL_004e:  br.s       IL_0058
+  IL_0050:  ldc.i4.4
+  IL_0051:  stloc.0
+  IL_0052:  br.s       IL_0058
+  IL_0054:  ldc.i4.5
+  IL_0055:  stloc.0
+  IL_0056:  br.s       IL_0058
+  IL_0058:  ldc.i4.1
+  IL_0059:  brtrue.s   IL_005c
+  IL_005b:  nop
+  IL_005c:  ldloc.0
+  IL_005d:  stloc.1
+  IL_005e:  br.s       IL_0060
+  IL_0060:  ldloc.1
+  IL_0061:  ret
+}");
+        }
+
+        [Fact, WorkItem(46536, "https://github.com/dotnet/roslyn/issues/46536")]
+        public void MultiplePathsToNode_SwitchDispatch_04()
+        {
+            var source = @"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        Console.Write(M(""a"", ""x"")); // 0
+        Console.Write(M(""a"", ""y"")); // 1
+        Console.Write(M(""a"", ""z"")); // 2
+        Console.Write(M(""a"", ""w"")); // 3
+        Console.Write(M(""b"", ""z"")); // 2
+        Console.Write(M(""c"", ""z"")); // 3
+        Console.Write(M(""c"", ""w"")); // 3
+    }
+
+    static int M(string x, string y)
+    {
+        return (x, y) switch
+        {
+            (""a"", ""x"") => 0,
+            (""a"", ""y"") => 1,
+            (""a"" or ""b"", ""z"") => 2,
+            _ => 3
+        };
+    }
+}
+";
+            var verifier = CompileAndVerify(source, expectedOutput: "0123233", options: TestOptions.DebugExe);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("Program.M", @"{
```

Comment:
I see that we didn't generate a switch dispatch in this scenario, so this looks like a different scenario than the one we looked at together. Could you explain what was happening in this case?

Reply:
We lower into a BoundSwitchDispatch, so the LocalRewriter code path we're interested in is exercised, but the BoundSwitchDispatch [is not sufficiently complex](https://github.com/dotnet/roslyn/blob/892f9fae716ffb7b6fbd4a32db1eeca8fd376f6f/src/Compilers/Core/Portable/CodeGen/SwitchStringJumpTableEmitter.cs#L225) for the compiler to decide to switch on the string hash code and it instead emits an if/else if/... chain.